sympy/functions/elementary/trigonometric.py

```diff
@@ -1826,13 +1828,16 @@ def taylor_term(n, x, *previous_terms):
             return S.NegativeOne**k*euler(2*k)/factorial(2*k)*x**(2*k)
 
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.calculus.accumulationbounds import AccumBounds
         arg = self.args[0]
         x0 = arg.subs(x, 0).cancel()
         n = (x0 + S.Pi/2)/S.Pi
         if n.is_integer:
             lt = (arg - n*S.Pi + S.Pi/2).as_leading_term(x)
             return (S.NegativeOne**n)/lt
-        return self.func(x0)
+        if x0 is S.ComplexInfinity:
+            return AccumBounds(S.NegativeInfinity, S.Infinity)
```

Comment:
AccumBounds is essentially extended real. It should not be used for expansions and limits at zoo.

Reply:
True, this slipped out of my mind . I'll fix that !